### ISSUE TITLE
A question about the test accuracy of BERT-based-uncased model on the MNLI dataset

### ISSUE THREAD

**Issue by 14H034160212** [huggingface/transformers#5272]
# 🐛 Bug
I am curious about how accuracy you guys can reach by using the BERT-based-uncased model on the MNLI task? I got 61% test accuracy on the MNLI-matched dataset and 62% accuracy on the MNLI-unmatched dataset.
## Information

Model I am using Bert-based-uncased:

Language I am using the model on English:

The problem arises when using:
* I use the exactly the same official example scripts: (give details below)
https://github.com/huggingface/transformers#quick-tour
https://github.com/huggingface/transformers#run_gluepy-fine-tuning-on-glue-tasks-for-sequence-classification

The tasks I am working on is:
* An official GLUE task: (MNLI)

## Environment info

     
- `transformers` version: v2.11.0
- Platform: Linux
- Python version: 3.7
- PyTorch version (GPU?): 1.5.0
- Using GPU in script?: 2 Tesla V100 GPUs
- Using distributed or parallel set-up in script?: No

**Comment by chaitanyamalaviya** [comment]
Same, I can't replicate the results in the README file.

**Comment by 14H034160212** [comment]
> Same, I can't replicate the results in the README file.

I also try to use BERT-large-uncased model and get 0.6209 eval_mnli/acc and 0.6281 eval_mnli-mm/acc.

**Comment by chaitanyamalaviya** [comment]
FYI, I can get 84.25 eval_mnli/acc by using a non-zero number of warmup steps. . 
Specifically, adding these lines in `trainer.py`:
```
warmup_steps = float(num_training_steps)*0.1
scheduler = get_linear_schedule_with_warmup(optimizer, num_warmup_steps=warmup_steps, num_training_steps=num_training_steps)
```

**Comment by 14H034160212** [comment]
> FYI, I can get 84.25 eval_mnli/acc by using a non-zero number of warmup steps. .
> Specifically, adding these lines in `trainer.py`:
> 
> ```
> warmup_steps = float(num_training_steps)*0.1
> scheduler = get_linear_schedule_with_warmup(optimizer, num_warmup_steps=warmup_steps, num_training_steps=num_training_steps)
> ```

Many thanks for your sharing. Cheers!

**Comment by 14H034160212** [comment]
> FYI, I can get 84.25 eval_mnli/acc by using a non-zero number of warmup steps. .
> Specifically, adding these lines in `trainer.py`:
> 
> ```
> warmup_steps = float(num_training_steps)*0.1
> scheduler = get_linear_schedule_with_warmup(optimizer, num_warmup_steps=warmup_steps, num_training_steps=num_training_steps)
> ```
Hi, Chaitanya I try to fix the code, but I got an error when running the program. Here is a double check that does the path of the ``trainer.py`` is "src/transformers/trainer.py"? And the lines you added are beginning from the line 312 in the ``trainer.py``?

**Comment by chaitanyamalaviya** [comment]
Adding a non-zero number of warmup steps in `src/transformers/trainer.py` is the only change I made. This error seems to occur because you have an output label that is outside the range [0, n_classes) while computing the loss. 
Did you make any other changes to the trainer/dataset? And are you training with the original MNLI dataset?
